contracts/pcv/EthPCVSwapperUniswap.sol

```diff
@@ -0,0 +1,145 @@
+        emit WithdrawERC20(msg.sender, to, token, amount);
+    }
+
+    /// @notice Sets the token to spend
+    /// @param _tokenSpent the address of the token to spend
+    function setTokenSpent(address _tokenSpent) external override onlyGovernor {
+        require(_tokenSpent == router.WETH(), "EthPCVSwapperUniswap: can only spend ETH");
+    }
+
+    /// @notice Sets the token to receive
+    /// @param _tokenReceived the address of the token to receive
+    function setTokenReceived(address _tokenReceived) external override onlyGovernor {
+      tokenReceived = _tokenReceived;
+      emit UpdateTokenReceived(_tokenReceived);
+    }
+
+    /// @notice Get the token to spend
+    /// @return The address of the token to spend
+    function getTokenSpent() external view override returns (address) {
+      return router.WETH();
+    }
+
+    /// @notice Get the token to receive
+    /// @return The address of the token to receive
+    function getTokenReceived() external view override returns (address) {
+      return tokenReceived;
+    }
+
+    function swap() external override afterTime whenNotPaused {
+      require(address(this).balance > 0, "EthPCVSwapperUniswap: no ETH left");
+      require(IERC20(tokenReceived).balanceOf(address(this)) < tokenBuyLimit, "EthPCVSwapperUniswap: cannot buy more");
+
+      // Get pair reserves
+      IWETH weth = IWETH(router.WETH());
+      (uint256 _token0, uint256 _token1) = getReserves();
+      (uint256 ethReserves, uint256 tokenReserves) =
+          pair.token0() == address(weth)
+              ? (_token0, _token1)
+              : (_token1, _token0);
+
+      // Prepare swap
+      uint256 balance = address(this).balance;
+      uint256 amount = Math.min(maxEthSpentPerSwap, balance);
+
+      uint256 amountOut =
+          UniswapV2Library.getAmountOut(amount, ethReserves, tokenReserves);
+
+      // Check spot price vs oracle price discounted by max slippage
+      // E.g. for a max slippage of 3% :
+      // spot price must be >= oraclePrice * (1 - 3%)
+      (Decimal.D256 memory twap, bool oracleValid) = oracle.read();
+      require(oracleValid, "EthPCVSwapperUniswap: invalid oracle");
+      uint256 oracleAmountOut = twap.mul(amount).asUint256();
+      Decimal.D256 memory maxSlippage = Decimal.ratio(SLIPPAGE_GRANULARITY - maximumSlippage, SLIPPAGE_GRANULARITY);
+      Decimal.D256 memory oraclePriceMinusSlippage = maxSlippage.mul(oracleAmountOut);
+      require(Decimal.lessThanOrEqualTo(oraclePriceMinusSlippage, Decimal.from(amountOut)), "EthPCVSwapperUniswap: slippage too high");
+
+      // Reset timer
+      _initTimed();
+
+      // Perform swap
+      weth.deposit{value: amount}();
+      assert(weth.transfer(address(pair), amount));
+
+      (uint256 amount0Out, uint256 amount1Out) =
+          pair.token0() == address(weth)
+              ? (uint256(0), amountOut)
+              : (amountOut, uint256(0));
+      pair.swap(amount0Out, amount1Out, address(this), new bytes(0));
```

Comment:
we could have the `to` address be a configurable `target` like the EthPCVDripper